app/src/main/java/edu/wisc/ece/pinpoint/pages/newpin/NewPinFragment.java

```diff
@@ -233,6 +246,19 @@ private void createNewPin() {
         });
     }
 
+    private void setPinniesDrawable(){
+        TypedValue colorValue = new TypedValue();
+        getContext().getTheme().resolveAttribute(com.google.android.material.R.attr.colorOnPrimary, colorValue, true);
+
+        Drawable unwrapped_pinnies_logo_button = AppCompatResources.getDrawable(getContext(), R.drawable.ic_pinnies_logo);
+        pinnies_logo_button = DrawableCompat.wrap(unwrapped_pinnies_logo_button);
+        DrawableCompat.setTint(pinnies_logo_button, colorValue.data);
+
+        Drawable unwrapped_pinnies_logo_title = AppCompatResources.getDrawable(getContext(), R.drawable.ic_pinnies_logo);
+        pinnies_logo_title = DrawableCompat.wrap(unwrapped_pinnies_logo_title);
+        DrawableCompat.setTint(pinnies_logo_title, colorValue.data);
+    }
```

Comment:
Same as above, is there a way to avoid having code just to put the drawable on the screen? If there's no way not to, could we make this a util function we can reuse?

Reply:
I'll probably keep this here for the button case only - the alternative to get the icon onto the button is to set an arbitrary amount of padding and just slap the icon over that spot. I can take the text-bar icon and throw that into the XML though

Reply:
If the icon is always a certain size (which I think it is?) you could just specify an amount of padding for the button if that would work. I think anything that reduces the amount of code we have is good since we don't want to have unnecessary stuff that could be confusing later.

Reply:
That's what I did to get the loading circle to appear on the button, so I guess it works well enough. Probably not the best design practice but oh well

Reply:
haha yeah probs not. Whatever you think is the best way to do it is good with me. I just think the less code we have to look through and potentially debug the better 😄